### PR TITLE
Split team building service types by usage

### PR DESCRIPTION
**Pull request by buoyad**
`ServiceIdWithContact` is the set of possibly selected tabs in the modal.
`ServiceId` is set of services we can pass to the search RPC.

Now, transform `ServiceIdWithContact` -> `ServiceId` in the team building container to call search.

### CODE REVIEW

File: shared/team-building/email-input/index.tsx [hunk 1]
```diff
@@ -26,6 +26,7 @@ const EmailInput = ({namespace}: EmailInputProps) => {
   const emailHasKeybaseAccount =
     user !== null && user.serviceMap.keybase !== '' && emailString === user.username
 
+  // TODO use search from container instead
```
Comment (buoyad) [inline shared/team-building/email-input/index.tsx]: Coming soon in pt.2

File: shared/actions/team-building.tsx [hunk 2]
```diff
@@ -61,8 +53,8 @@ function* searchResultCounts(state: TypedState, {payload: {namespace}}: NSAction
 
   // Filter on `services` so we only get what's searchable through API.
   // Also filter out if we already have that result cached.
-  const servicesToSearch = Constants.services
-    .filter(s => s !== teamBuildingSelectedService && !['contact', 'phone', 'email'].includes(s))
+  const servicesToSearch = Constants.allServices
```
Comment (buoyad) [inline shared/actions/team-building.tsx]: Removing in pt.3

**Comment by buoyad** [comment]
cc @keybase/y2ksquad @keybase/picnicsquad this is ready

File: shared/actions/team-building.tsx (hunk 2 above)
Comment (buoyad) [reply shared/actions/team-building.tsx]: https://github.com/keybase/client/pull/19299

File: shared/team-building/email-input/index.tsx (hunk 1 above)
Comment (buoyad) [reply shared/team-building/email-input/index.tsx]: https://github.com/keybase/client/pull/19296

File: shared/constants/team-building.tsx [hunk 3]
```diff
@@ -172,15 +178,3 @@ const interestingPersonToUser = (person: RPCTypes.InterestingPerson): Types.User
     username: username,
   }
 }
-
-export {
```
Comment (MarcoPolo) [inline shared/constants/team-building.tsx]: ? Is this a new pattern we're following now?

Comment (chrisnojima) [reply shared/constants/team-building.tsx]: I think so. It just adds a bunch of lines for not much gain

Comment (MarcoPolo) [reply shared/constants/team-building.tsx]: Great I'm all for it.

**Pull request merged**